session/manager.go

```diff
@@ -44,30 +62,58 @@ func (m *Manager) Driver(name ...string) (sessioncontract.Driver, error) {
 	}
 
 	if m.drivers[driverName] == nil {
-		if m.customDrivers[driverName] == nil {
-			return nil, fmt.Errorf("driver [%s] not supported", driverName)
-		}
-
-		m.drivers[driverName] = m.customDrivers[driverName]
+		return nil, fmt.Errorf("driver [%s] not supported", driverName)
 	}
 
 	return m.drivers[driverName], nil
 }
 
-func (m *Manager) Extend(driver string, handler func() sessioncontract.Driver) sessioncontract.Manager {
-	m.customDrivers[driver] = handler()
-	return m
+func (m *Manager) Extend(driver string, handler sessioncontract.Driver) (sessioncontract.Manager, error) {
+	if m.drivers[driver] != nil {
+		return nil, fmt.Errorf("driver [%s] already exists", driver)
+	}
+	m.drivers[driver] = handler
+	m.startGcTimer(m.drivers[driver])
```

Comment:
How will this `GcTimer` start for the built-in driver in Goravel?

Reply:
> How will this `GcTimer` start for the built-in driver in Goravel?

```
func (m *Manager) extendDefaultDriver() {
	if _, err := m.Extend("file", m.createFileDriver); err != nil {
		panic(fmt.Sprintf("failed to extend session manager: %v", err))
	}
}

func (m *Manager) createFileDriver() sessioncontract.Driver {
	lifetime := m.config.GetInt("session.lifetime")
	return driver.NewFile(m.config.GetString("session.files"), lifetime)
}
```

built-in driver also use `Extend` to register, so GC will start when register the driver.

Reply:
Oh, okay, sorry, I didn't see the updated code. But how will we handle multiple built-in drivers?

Reply:
> Oh, okay, sorry, I didn't see the updated code. But how will we handle multiple built-in drivers?

maybe:

```
func (m *Manager) extendDefaultDrivers() {
	if _, err := m.Extend("file", m.createFileDriver); err != nil {
		panic(fmt.Sprintf("failed to extend session file driver: %v", err))
	}
	if _, err := m.Extend("file", m.createDatabaseDriver); err != nil {
		panic(fmt.Sprintf("failed to extend session database driver: %v", err))
	}
}

func (m *Manager) createFileDriver() sessioncontract.Driver {
	lifetime := m.config.GetInt("session.lifetime")
	return driver.NewFile(m.config.GetString("session.files"), lifetime)
}

func (m *Manager) createDatabaseDriver() sessioncontract.Driver {
	// TODO
}

But I think only need to implement file driver for default, users may not use database, if they need database driver they can add it manualy.
```

Reply:
This is controlled by `config/session.go`. So if the user has the `driver` set to the `database`, we should extend the database driver.